AutoReplayUploader/AutoReplayUploaderPlugin.cpp

```diff
@@ -5,194 +5,341 @@
-	//cvarManager->registerCvar("cl_autoreplayupload_calculated_endpoint", CALCULATED_ENDPOINT_DEFAULT, "URL to upload replay to when uploading to calculated.gg instance");
-
-	/*
-	Load notification assets
-	*/
-	gameWrapper->LoadToastTexture("calculated_logo", "./bakkesmod/data/assets/calculated_logo.tga");
-	gameWrapper->LoadToastTexture("ballchasing_logo", "./bakkesmod/data/assets/ballchasing_logo.tga");
-	cvarManager->registerCvar("cl_autoreplayupload_notifications", "1", "Show notifications on successful uploads", true, true, 0, true, 1).bindTo(showNotifications);
-
+	steamUserName = steamClient->GetISteamFriends(steamUser, steamPipe, "SteamFriends017")->GetPersonaName();
 }
 
+/**
+* OnUnload event called when the plugin is unloaded by BakkesMod
+*/
 void AutoReplayUploaderPlugin::onUnload()
 {
 }
 
+/**
+* OnGameComplete event called when on Function TAGame.GameEvent_Soccar_TA.EventMatchEnded event when an online game ends.
+* Params:
+*	caller - ServerWraper this event was called from
+*	params - Event parameters
+*	eventName - Event name
+*/
 void AutoReplayUploaderPlugin::OnGameComplete(ServerWrapper caller, void * params, std::string eventName)
 {
-	if (!*uploadToCalculated && !*uploadToBallchasing)
+	if (!*uploadToCalculated && !*uploadToBallchasing) // Bail if we aren't uploading replays
 	{
 		return; //Not uploading replays
 	}
+
+	// Get ReplayDirector
 	ReplayDirectorWrapper replayDirector = caller.GetReplayDirector();
 	if (replayDirector.IsNull())
 	{
 		cvarManager->log("Could not upload replay, director is NULL!");
-		if(*showNotifications) gameWrapper->Toast("Autoreplayuploader", "Error exporting replay! (1)", "default", 3.5f, ToastType_Error);
+#ifdef TOAST
+		if (*showNotifications) gameWrapper->Toast("Autoreplayuploader", "Error exporting replay! (1)", "default", 3.5f, ToastType_Error);
+#endif
 		return;
 	}
+
+	// Get Replay wrapper
 	ReplaySoccarWrapper soccarReplay = replayDirector.GetReplay();
 	if (soccarReplay.memory_address == NULL)
 	{
 		cvarManager->log("Could not upload replay, replay is NULL!");
+#ifdef TOAST
 		if (*showNotifications) gameWrapper->Toast("Autoreplayuploader", "Error exporting replay! (2)", "default", 3.5f, ToastType_Error);
+#endif
 		return;
 	}
-	std::string replayPath = "./bakkesmod/data/autoreplaysave.replay";// cvarManager->getCvar("cl_autoreplayupload_filepath").getStringValue(); //"./bakkesmod/data/autoreplaysave.replay";//
+
+	// If we have a template for the replay name then set the replay name based off that template
+	auto replayNameTemplate = cvarManager->getCvar(CVAR_REPLAY_NAME_TEMPLATE).getStringValue();
+	if (!replayNameTemplate.empty())
+	{
+		SetReplayName(caller, soccarReplay, replayNameTemplate);
+	}
+
+	// Determine the replay path and if it already exists remove it
+	std::string replayPath = "./bakkesmod/data/autoreplaysave.replay";// cvarManager->getCvar(CVAR_REPLAY_UPLOAD_PATH).getStringValue();
 	if (file_exists(replayPath))
```

Comment:
After this change I'd like to make this finish the parameterization of the replay export path so I can choose to have all my replays exported locally somewhere for later importing to the demo folder.

Reply:
Sounds good to me. A feature like save last X replays has been suggested a few times as well, maybe that can be tied into this?

Reply:
Hm... interesting I suppose we could add a variable to the config that would say something like save last 10 replays and if not set it just tries to save everything.  Still would have the problem of name collisions which I'm noodling on how best to solve now.

My current thought with collisions is take the name of the replay and use that for the filename and if it already exists then it is the user's choice because they made it that way with the replay template.  If they don't use a replay template then the filename will be the ID of the replay like before and shouldn't get overridden.

Reply:
A couple of random alphanums at the end of the replay name would probably avoid name collisions. Like:
`{rendered_user_filename_from_template}-{three_random_alphanums}.replay` or throw on ` (2)`, `(3)` etc if there is collisions like windows does.

/shrug